src/Resources/Timeline.php

```diff
@@ -0,0 +1,262 @@
+<?php
+
+namespace SevenShores\Hubspot\Resources;
+
+use DateTime;
+
+class Timeline extends Resource
+{
+    /**
+     * Create or Update Timeline Event
+     *
+     * @param int           $appId
+     * @param int           $eventTypeId
+     * @param string        $id
+     * @param int|null      $objectId
+     * @param string|null   $email
+     * @param string|null   $utk
+     * @param array         $extraData
+     * @param DateTime|null $timestamp
+     *
+     * @return mixed
+     *
+     * @see http://developers.hubspot.com/docs/methods/timeline/create-or-update-event
+     */
+    public function createOrUpdate(
+        $appId,
+        $eventTypeId,
+        $id,
+        $objectId = null,
+        $email = null,
+        $utk = null,
+        $extraData = [],
+        DateTime $timestamp = null
+    ) {
+        $endpoint = "https://api.hubapi.com/integrations/v1/{$appId}/timeline/event";
+
+        $data['json'] = [
+            'eventTypeId' => $eventTypeId,
+            'id'          => $id,
+            'objectId'    => $objectId,
+            'email'       => $email,
+            'utk'         => $utk,
+            'extraData'   => $extraData,
+        ];
+
+        if (isset($timestamp)) {
+            // Times must be in milliseconds epoch time.
+            $data['json']['timestamp'] = $timestamp->getTimestamp() * 1000;
+        }
```

Comment:
I think with a variable name of `$timestamp` people should know that we are expecting an integer timestamp and we don't need to muddy up the code with a bunch of checks.

If it really is a concern, maybe add a timestamp method like this in the Resource base class:

```php
public function timestamp($timestamp)
{
    switch (true) {
        case: $timestamp instanceof \DateTime:
            return $timestamp->getTimestamp() * 1000;
        case: is_numeric($timestamp) && strlen((string)$timestamp) == 10:
            return $timestamp * 1000;
        default:
            return $timestamp;
        }
    }
}
```

then you can just do:

```php
$data['json']['timestamp'] = $this->timestamp($timestamp);
```